components/tikv_alloc/src/jemalloc.rs

```diff
@@ -5,25 +5,80 @@ use crate::AllocStats;
 use jemalloc_ctl::{stats, Epoch as JeEpoch};
 use jemallocator::ffi::malloc_stats_print;
 use libc::{self, c_char, c_void};
-use std::{io, ptr, slice};
+use std::collections::HashMap;
+use std::{io, ptr, slice, sync::Mutex, thread};
 
 pub type Allocator = jemallocator::Jemalloc;
 pub const fn allocator() -> Allocator {
     jemallocator::Jemalloc
 }
 
+lazy_static! {
+    static ref THREAD_MEMORY_MAP: Mutex<HashMap<ThreadId, MemoryStatsAccessor>> =
+        Mutex::new(HashMap::new());
+}
+
+struct MemoryStatsAccessor {
+    allocated: jemalloc_ctl::thread::AllocatedP,
+    deallocated: jemalloc_ctl::thread::DeallocatedP,
+    thread_name: String,
+}
+
+pub fn add_thread_memory_accessor() {
+    let mut thread_memory_map = THREAD_MEMORY_MAP.lock().unwrap();
```

Comment:
I'm not sure whether it is better to use a concurrent hash map instead.

Reply:
Since the frequency of change is low (only human query and thread initialization step), I guess this is ok.